lib/http-invocation.js

```diff
@@ -196,18 +196,24 @@ HttpInvocation.prototype.createRequest = function() {
 
   // add auth if it is set
   if (auth) {
-    req.auth = {};
     if (auth.username && auth.password) {
+      req.auth = {};
       req.auth.username = auth.username;
       req.auth.password = auth.password;
     }
     if (auth.bearer) {
+      req.auth = req.auth || {};
       req.auth.bearer = auth.bearer;
     }
-    if ('sendImmediately' in auth) {
-      req.auth.sendImmediately = auth.sendImmediately;
-    } else {
-      req.auth.sendImmediately = false;
+    if (req.auth) {
+      if ('sendImmediately' in auth) {
+        req.auth.sendImmediately = auth.sendImmediately;
+      } else {
+        req.auth.sendImmediately = false;
+      }
+    } else if (auth.accessToken) {
+      req.headers = req.headers || {};
+      req.headers.Authorization = auth.accessToken.id;
```

Comment:
I find this structure difficult to comprehend, there are too many different code paths possible. I reworked it by moving the branch where we are setting `req.headers` to the top and preserving the old code mostly intact.